controllers/capabilities/tenant_internal_reconciler.go

```diff
@@ -104,54 +100,56 @@ func (r *TenantInternalReconciler) reconcileTenant() (ctrl.Result, error) {
 
 		updated, err := r.reconcileStatus(newStatus)
 		if err != nil {
-			return ctrl.Result{}, err
+			return false, err
 		}
 
+		// If updated - update the status and requeue
 		if updated {
-			// requeue to have a new run with the updated tenant resource
-			return ctrl.Result{Requeue: true}, nil
+			return true, nil
 		}
 	}
 
-	return ctrl.Result{}, nil
+	return false, nil
 }
 
 // This method makes sure admin user:
 // * is active
 // * user's attributes will be updated if required
-func (r *TenantInternalReconciler) reconcileAdminUser() (ctrl.Result, error) {
+func (r *TenantInternalReconciler) reconcileAdminUser() error {
 	tenantID, err := r.retrieveTenantID()
 	if err != nil {
-		return ctrl.Result{}, errors.New("failed to convert tenantID annotation to int64")
+		return errors.New("failed to convert tenantID annotation to int64")
 	}
 
 	if tenantID == 0 {
-		return ctrl.Result{}, errors.New("trying to reconcile admin user when tenantID 0")
+		return errors.New("trying to reconcile admin user when tenantID 0")
 	}
 
 	adminUser, err := controllerhelper.FindUser(r.portaClient, tenantID,
 		r.tenantR.Spec.Email, r.tenantR.Spec.Username)
 	if err != nil {
-		return ctrl.Result{}, err
+		return err
 	}
 
 	if adminUser == nil {
 		adminUser, err = controllerhelper.CreateAdminUser(r.portaClient,
 			tenantID, r.tenantR.Spec.Email, r.tenantR.Spec.Username)
 		if err != nil {
-			return ctrl.Result{}, err
+			return &helper.WaitError{
+				Err: fmt.Errorf("3scale client failed creating the admin user: %v", err),
```

Comment:
Could be left as it was - however I found that some of the error responses from 3scale api aren't suitable to be displayed on the CR